src/tribler/core/upgrade/version_manager.py

```diff
@@ -72,14 +72,14 @@ class TriblerVersion:
     should_recreate_directory: bool
     deleted: bool
 
-    def __init__(self, root_state_dir: Path, version_str: str, files_to_copy: List[str],
+    def __init__(self, root_state_dir: Path, version_str: str, files_to_copy: List[str] = None,
                  last_launched_at: Optional[float] = None):
         if last_launched_at is None:
             last_launched_at = time.time()
         self.logger = logging.getLogger(self.__class__.__name__)
         self.version_str = version_str
-        self.version_tuple = tuple(LooseVersion(version_str).version)
-        self.major_minor = self.version_tuple[:2]
+        self.version = LooseVersion(version_str)
+        self.major_minor = tuple(self.version.version[:2])
```

Comment:
**nitpick:** `self.version.version[:2]` should already be a tuple, so it is not necessary to use `tuple(...)` here

```suggestion
        self.major_minor = self.version.version[:2]
```

Reply:
`self.version.version` is a list:

```python
class LooseVersion (Version):
    component_re = re.compile(r'(\d+ | [a-z]+ | \.)', re.VERBOSE)

    def __init__ (self, vstring=None):
        if vstring:
            self.parse(vstring)


    def parse (self, vstring):
        self.vstring = vstring
        components = [x for x in self.component_re.split(vstring)
                              if x and x != '.']
        for i, obj in enumerate(components):
            try:
                components[i] = int(obj)
            except ValueError:
                pass

        self.version = components
```